### PR TITLE
Add test_replica_failure_during_attaching test

### PR DESCRIPTION
**Pull request by khushboo-rancher**
https://github.com/longhorn/longhorn/issues/1999

### CODE REVIEW

**Review by shuo-wu — CHANGES_REQUESTED** [review]
The test steps need to be updated:
```
10. Wait for the replenishment interval.
11. Enable the default disk for the host node.	
12. Wait for volume2 to become Healthy. 
        --> Verify the failed replica using the extra disk will not be reused. Instead, a new replica will be rebuilt in the default disk enabled in step11
13. Verify data content and r/w capability for volume2.
```

File: manager/integration/tests/test_ha.py [hunk 1]
```diff
@@ -2370,4 +2359,68 @@ def test_replica_failure_during_attaching():
...
+
+    update_disks = get_update_disks(node.disks)
+    default_disk_name = next(iter(update_disks))
+    update_disks["extra-disk"] = extra_disk
+    update_disks[default_disk_name].allowScheduling = False
+
+    node = node.diskUpdate(disks=update_disks)
+    node = common.wait_for_disk_update(client, node.name, len(update_disks))
+
+    volume_name_2 = volume_name_1 + '-2'
```
[the remark this answers is not in the document]
Comment (meldafrawi) [reply manager/integration/tests/test_ha.py:2387]: fixed

File: manager/integration/tests/test_ha.py [hunk 2]
```diff
@@ -2370,4 +2359,68 @@ def test_replica_failure_during_attaching():
...
+    volume_2 = create_and_check_volume(client, volume_name_2, 3, str(1 * Gi))
+    volume_2.attach(hostId=host_id)
+    volume_2 = common.wait_for_volume_healthy(client, volume_name_2)
+    write_volume_random_data(volume_2)
+    volume_2.detach()
+    wait_for_volume_detached(client, volume_name_2)
+
+    # unmount the disk
+    mount_path = os.path.join(DIRECTORY_PATH, volume_name_1)
+    common.lazy_umount_disk(mount_path)
```
[the remark this answers is not in the document]
Comment (meldafrawi) [reply manager/integration/tests/test_ha.py:2397]: fixed

**Comment by meldafrawi** [comment]
@shuo-wu I've fixed all your comments and added them to a different PR https://github.com/longhorn/longhorn-tests/pull/531

@khushboo-rancher Great job, Thank you.

**Comment by meldafrawi** [comment]
Closing this PR in favor to https://github.com/longhorn/longhorn-tests/pull/531

**Pull request closed without merging**